core/engine/src/spanned_source_text.rs

```diff
@@ -0,0 +1,52 @@
+use boa_ast::{LinearSpan, SourceText};
+use boa_gc::{Finalize, Gc, Trace};
+
+#[derive(Trace, Finalize)]
+pub(crate) struct SourceTextInner {
+    #[unsafe_ignore_trace]
+    source_text: SourceText,
+}
```

Comment:
I would suggest we rename this to `SourceText` and avoid the naming conflict, by referencing `boa_ast::SourceText` directly where we need it.
Additionally I think this should probably have an `Inner` struct that contains the `boa_ast::SourceText` and is wrapped in `Gc`. Because everywhere this is used, I think it is wrapped in a `Gc` anywas and like this we just make it easier to work with.